src/lib/token_list_gen.ts

```diff
@@ -76,106 +127,203 @@ export const generateTokenList = async (
+  const { l1 , l2 } = await promiseErrorMultiplier(getNetworkConfig(), (error) => getNetworkConfig(), 3, false);
 
-  let tokens =
+  let tokens: GraphTokenResult[] =
     options && options.getAllTokensInNetwork
-      ? await getAllTokens(l2.network.chainID)
-      : await getTokens(
+      ?  await promiseErrorMultiplier(getAllTokens(l2.network.chainID), (error) => getAllTokens(l2.network.chainID), 3, false)
+      : await promiseErrorMultiplier(
+        getTokens(
           l1TokenList.tokens.map((token) => ({
             addr: token.address.toLowerCase(),
             logo: token.logoURI
           })),
           l2.network.chainID
-        );
-
+        ),
+        (error) => getTokens(
+          l1TokenList.tokens.map((token) => ({
+            addr: token.address.toLowerCase(),
+            logo: token.logoURI
+          })),
+          l2.network.chainID
+        ), 
+        3, 
+        false);
+  
+  const l1TokenAddresses =
+    options && options.getAllTokensInNetwork
+      ? tokens.map((curr) => curr.l1TokenAddr)
+      : l1TokenList.tokens.map((token) => token.address);
+
+  let intermediatel2AddressesFromL1 = [];
+  let intermediatel2AddressesFromL2 = [];
+  for (const addrs of getChunks(l1TokenAddresses)){
+      let l2AddressesFromL1Temp = await promiseErrorMultiplier(getL2TokenAddressesFromL1(addrs, l1.multiCaller, l2.network.tokenBridge.l1GatewayRouter), (error) => getL2TokenAddressesFromL1(addrs, l1.multiCaller, l2.network.tokenBridge.l1GatewayRouter), 3, false);
+      intermediatel2AddressesFromL1.push(l2AddressesFromL1Temp);
+      let l2AddressesFromL2Temp = await promiseErrorMultiplier(getL2TokenAddressesFromL2(addrs, l2.multiCaller, l2.network.tokenBridge.l2GatewayRouter), (error) => getL2TokenAddressesFromL2(addrs, l2.multiCaller, l2.network.tokenBridge.l2GatewayRouter), 3, false);
+      intermediatel2AddressesFromL2.push(l2AddressesFromL2Temp);
+  }
+  const l2AddressesFromL1 = intermediatel2AddressesFromL1.flat(1);
+  const l2AddressesFromL2 = intermediatel2AddressesFromL2.flat(1);
   
-  const l1TokenAddresses = tokens.map((token:GraphTokenResult) => token.l1TokenAddr);
-  const l2AddressesFromL1 = await getL2TokenAddressesFromL1(l1TokenAddresses, l1.multiCaller, l2.network.tokenBridge.l1GatewayRouter);
-  const l2AddressesFromL2 = await getL2TokenAddressesFromL2(l1TokenAddresses, l2.multiCaller, l2.network.tokenBridge.l2GatewayRouter);
+  if(isNova) {
+    const logos = l1TokenList.tokens.reduce(
+      (acc, curr) => ((acc[curr.address.toLowerCase()] = curr.logoURI), acc),
+      {} as { [addr: string]: string | undefined }
+    );
+    
+    let intermediatel2GatewayAddrs = [];
+    for (const addrs of getChunks(l1TokenAddresses)){
+      let l2GatewayAddressesFromL1Temp = await promiseErrorMultiplier(getL2GatewayAddressesFromL1Token(addrs, l2.multiCaller, l2.network), (error) => getL2GatewayAddressesFromL1Token(addrs, l2.multiCaller, l2.network), 3, false);
```

Comment:
would be good to break this down into multiple lines (or use prettier formatter to lint it)